go-selinux/selinux_linux.go

```diff
@@ -935,12 +944,13 @@ func loadLabels() (map[string]string, string) {
 	con, _ := NewContext(labels["file"])
 	con["level"] = fmt.Sprintf("s0:c%d,c%d", maxCategory-2, maxCategory-1)
 	reserveLabel(con.get())
-	return labels, con.get()
+	privContainerMountLabel = con.get()
 }
 
 // kvmContainerLabels returns the default processLabel and mountLabel to be used
 // for kvm containers by the calling process.
 func kvmContainerLabels() (string, string) {
+	loadLabels()
 	processLabel := labels["kvm_process"]
 	if processLabel == "" {
```

Comment:
I see most (all?) places we call `loadLabels()` are followed by `something := labels[...]`. Having to call `loadLabels()` is something that _could_ be forgotten.

What if we make this more "natural", and instead of calling `loadLabels()` directly, implement it as an accessor for `labels`, something like:

```go
func getLabel(name string) string {
	loadLabelsOnce.Do(func() {
		doLoadLabels()
	})
	return labels[name]
}
```

Then this could be written as

```suggestion
	processLabel := getLabel("kvm_process")
  	if processLabel == "" {
```

Reply:
Yes, makes sense (except the `PrivContainerMountLabel` looks a tad odd, but I've added a comment).